### ISSUE TITLE
Add SELU (Scaled ELU)

### ISSUE THREAD

**Issue by ethancaballero** [pytorch/pytorch#1768]
Here's TF example:
https://github.com/bioinf-jku/SNNs/blob/master/selu.py#L20-L25

Ideally it would be done in C in TH backend.
 
However, it could also be done in python if a pytorch equivalent of np.where was available:
https://discuss.pytorch.org/t/how-can-i-do-the-operation-the-same-as-np-where/1329/3

**Comment by fmassa** [comment]
It looks like it can be implemented somewhat efficiently in one line:
```python
import torch.nn.functional as F
def selu(x):
    alpha = 1.6732632423543772848170429916717
    scale = 1.0507009873554804934193349852946
    return scale * F.elu(x, alpha)
```
Note that pytorch `elu` has the `alpha` parameter built-in, which makes it easier to implement than in TF. For an overview of what the `alpha` parameter do, you can have a [look here](https://github.com/pytorch/pytorch/blob/master/torch/lib/THNN/generic/ELU.c#L23).

**Comment by ethancaballero** [comment]
add "Alpha dropout" too:
https://github.com/bioinf-jku/SNNs/blob/master/selu.py#L32-L68

so that SELU is still effective with dropout

**Comment by fmassa** [comment]
@ethancaballero I'll add Alpha dropout in a new PR.

**Comment by dannysdeng** [comment]
I've tried implementing selu and the dropout variant by borrowing heavily from the tf implementation:
https://github.com/dannysdeng/selu/blob/master/selu.py

**Comment by yxchng** [comment]
@fmassa Can I know how I can implement np.where in pytorch? I cannot figure out an efficient way to do it. (not for SELU)

**Comment by fmassa** [comment]
@yxchng questions like this are better suited for https://discuss.pytorch.org/
But I'd use a combination of indexing, `result[idx] = a[idx]` and `result[~idx] = b[~idx]`.

**Comment by owang** [comment]
@dannysdeng what about the weight initialization? This was an important part of the original paper for the theory about SELU to be correct from what I understand.

**Comment by rasbt** [comment]
Yes, you would initialize the weights via a normal distribution with mean 0 and std_dev=1/sqrt(num_inputs).

For example,

    self.conv_layer.weight.data.normal_(0.0, np.sqrt(1/(kernel_width*kernel_height)))
